remove dependencies on old Hydra texture system from hdMaya for USD releases after 20.11

mattyjams opened the pull request:
Our Hydra team is working on removing parts of the old Hydra texture system for the next core USD release.

With Luma's recent work, it looks like all of the necessary support for the newer material network-based system is already in place, so the changes here mostly just pre-process the old code away for USD releases after 20.11.

mattyjams commented:
@pmolodo: Could I bug you to have a look at this one when you get a chance? Just want to make sure we haven't missed anything here, since we don't make heavy use of mtoh ourselves.

Thanks!

pmolodo commented on lib/usd/hdMaya/adapters/materialAdapter.cpp:
```diff
@@ -105,7 +108,19 @@ void HdMayaMaterialAdapter::Populate()
+    return {};
+}
+
+#elif USD_VERSION_NUM <= 1911
```
this elif branch will never be entered - anything <= 1911 is by definition also <= 2011, so the first branch would be entered.

Either the order needs to be reversed (first check <= 1911, then elif <= 2011), or the first if check needs to be changed to:

```
#if USD_VERSION_NUM <= 2011 && USD_VERISON_NUM > 1911
```

mattyjams replied:
Ugh, exactly right, great catch. I'd protest that these weren't my changes, but I did review them and I completely glossed over that. :/

I'll fix these up!

mattyjams replied:
Addressed by d9ea7b2abc0f27f79638e3602c129a9812e73fd1.

pmolodo commented on lib/usd/hdMaya/adapters/materialAdapter.cpp:
```diff
@@ -317,7 +325,29 @@ class HdMayaShadingEngineAdapter : public HdMayaMaterialAdapter
+        return {};
+    }
+
+#elif USD_VERSION_NUM <= 1911
```
this elif branch will never be entered - see above note

mattyjams replied:
Addressed by d9ea7b2abc0f27f79638e3602c129a9812e73fd1.

pmolodo commented on lib/usd/hdMaya/adapters/materialAdapter.h:
```diff
@@ -43,7 +43,14 @@ class HdMayaMaterialAdapter : public HdMayaAdapter
+    HDMAYA_API
+    virtual HdTextureResource::ID GetTextureResourceID(const TfToken& paramName);
+
+#elif USD_VERSION_NUM <= 1911
```
this elif branch will never be entered - see above note

mattyjams replied:
Addressed by d9ea7b2abc0f27f79638e3602c129a9812e73fd1.

pmolodo reviewed (changes requested):
So, I haven't looked at the texture changes coming in 21.02 yet - is it the case that texture resource handling is handled more "automatically", and none of these overrides are required at all in order to get texture support?  Or are these changes effectively just disabling texture support for now (until we can add it back in a way that's compatible with 21.02, later?)

mattyjams commented:
> So, I haven't looked at the texture changes coming in 21.02 yet - is it the case that texture resource handling is handled more "automatically", and none of these overrides are required at all in order to get texture support? Or are these changes effectively just disabling texture support for now (until we can add it back in a way that's compatible with 21.02, later?)

I think it should be the former. I don't actually expect that there will be major changes needed to **add** support for 21.02, but just that vestiges of the old texture system are being removed, so we want to drop code in `hdMaya` that used those. The "new" system I was alluding to is actually in 20.11 now, where the scene delegate constructs an `HdMaterialNetworkMap` and returns that from `GetMaterialResource()`:

https://github.com/PixarAnimationStudios/USD/blob/release/pxr/imaging/hd/sceneDelegate.h#L598

It looks like all of that is already in place in `hdMaya` now. The actual handling of the texture resources should indeed now be handled auto-magically within Hydra itself based on the information in the material network.

So my hope is that texture support should happily continue working, but I was hoping you could help verify that. :)

pmolodo commented:
Oh - if the changes are in 20.11, then why are most of the checks for <= 20.11? Shouldn't they be < 20.11?

mattyjams commented:
> Oh - if the changes are in 20.11, then why are most of the checks for <= 20.11? Shouldn't they be < 20.11?

Good question. I don't think there's any particular reason why the deprecated API was left in for `== 20.11` other than that the Hydra team is midway through removing the old API and it made for a somewhat cleaner break. Would you prefer that I change it to `< 20.11` instead?

In any case, I rebased these changes and fixed up the `#if`/`#elif` stuff you noted, so thanks again for that! One additional change I added drops out the now-unused debug codes that go along with the removed `GetTextureResource()` and `GetTextureResourceID()`.

pmolodo commented:
If there's no benefit to supporting both APIs, then I think we should drop the old one in 20.11.  That will make the code read better, mean the old stuff will get dropped sooner, and also make it easier for me to test (since I already have a pre-compiled 20.11 at home, but haven't compiled the latest tip of dev yet!)

mattyjams commented:
> If there's no benefit to supporting both APIs, then I think we should drop the old one in 20.11. That will make the code read better, mean the old stuff will get dropped sooner, and also make it easier for me to test (since I already have a pre-compiled 20.11, but haven't compiled the latest tip of dev yet!)

Cool, works for me! I just rebased and pushed that change.

Let me know how your testing goes! Thanks!

pmolodo commented:
Ok, I built this and ran some basic tests, and it seems to be working fine!

pmolodo reviewed (approved):
Looks good!

mattyjams commented:
@pmolodo: Great, glad to hear it! Thank you, sir!
